leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/lockstep/LockStepTest.java

```diff
@@ -428,4 +434,47 @@ public void register_deregister_observe(String givenServerEndpointProvider) thro
         Set<Observation> observations = server.getObservationService().getObservations(registration);
         assertThat(observations).isEmpty();
     }
+
+    @TestAllTransportLayer
+    public void read_timestamped(String givenServerEndpointProvider) throws Exception {
+
+        // -------------------------------------------REGISTER
+        // CLIENT----------------------------------------------------
+        LockStepLwM2mClient client = new LockStepLwM2mClient(server.getEndpoint(Protocol.COAP).getURI());
+        Token token = client
+                .sendLwM2mRequest(new RegisterRequest(client.getEndpointName(), 60l, "1.1", EnumSet.of(BindingMode.U),
+                        null, null, linkParser.parseCoreLinkFormat("</1>,</2>,</3>".getBytes()), null));
+        client.expectResponse().token(token).go();
+        server.waitForNewRegistrationOf(client.getEndpointName());
+        Registration registration = server.getRegistrationService().getByEndpoint(client.getEndpointName());
+        // --------------------------------------------------------------------------------------------------------------
+
+        // ----------------------------------------------TIMESTAMP-------------------------------------------------------
+        LwM2mEncoder encoder = new DefaultLwM2mEncoder();
+        TimestampedLwM2mNodes.Builder builder = new TimestampedLwM2mNodes.Builder();
+        Instant t1 = Instant.now();
+        builder.put(t1, new LwM2mPath("/1/0/1"), LwM2mSingleResource.newIntegerResource(1, 3600));
+        TimestampedLwM2mNodes timestampedNodes = builder.build();
+        byte[] payload = encoder.encodeTimestampedNodes(timestampedNodes, ContentFormat.SENML_JSON,
+                client.getLwM2mModel());
+        // --------------------------------------------------------------------------------------------------------------
+
+        // Send read REQUEST
+        Future<ReadResponse> future = Executors.newSingleThreadExecutor().submit(new Callable<ReadResponse>() {
+            @Override
+            public ReadResponse call() throws Exception {
+                // send a request with 3 seconds timeout
+                return server.send(registration, new ReadRequest(ContentFormat.SENML_JSON, 1), 2000);
+            }
+        });
+
+        client.expectRequest().storeToken("TKN").storeMID("MID").go();
+
+        client.sendResponse(Type.ACK, ResponseCode.CONTENT, ContentFormat.SENML_JSON).loadMID("MID").loadToken("TKN")
+                .payload(payload).go();
+
+        ReadResponse response = future.get(2000, TimeUnit.MILLISECONDS);
```

Comment:
2 seconds timeout is maybe too high ? 1 is enough

(why not just using 1 and TimeUnit.SECONDS)